### ISSUE TITLE
modal bug: inputs should span the full width

### ISSUE THREAD

**Issue by aagonzales** [carbon-design-system/carbon#8373]
This is a regression: the inputs should be spanning the full width of the modal. 

Storybook:
![image](https://user-images.githubusercontent.com/11670886/114457646-e17d6c80-9ba3-11eb-99a8-19ce51f50cd1.png)

Should be:
![image](https://user-images.githubusercontent.com/11670886/114458210-86984500-9ba4-11eb-856b-c9844cebdf16.png)

**Comment by tw15egan** [comment]
@aagonzales we have a `hasForm` prop that allows things to span the full-width of the `Modal`, but it will also let the paragraph above go full width. 

The problem I think we're running into is that we are allowing consumers to render whatever they want inside of a modal, and the modal body has a varying padding-right based on the size of the screen. Since the inputs are also put inside the Modal body, they are affected by this as well (unless the `hasForm` prop has been provided). But like I pointed out, that will allow everything to go full width. 

Is the main reason for the right padding to constrain full-width paragraphs? If so, I wonder if we can just set the padding on all paragraph elements inside of the `Modal`. [We already set the font styles for all paragraphs](https://github.com/carbon-design-system/carbon/blob/main/packages/components/src/components/modal/_modal.scss#L269), so we would just be moving the padding from the modal body container to just the paragraphs.
